### ISSUE TITLE
Bug: failed load remote branches

### ISSUE THREAD

**Issue by wesharper** [gitbutlerapp/gitbutler#3129]
I keep getting this for my main gitlab repo: 
![image](https://github.com/gitbutlerapp/gitbutler/assets/13283469/7c4bcaf9-9d42-43f3-afa6-14b83b1b27b2)

I've used the `Test credentials` button in the project settings and everything passes. Currently on version `0.10.24`. Mac M2 Pro on Sonoma 14.4.

**Comment by mtsgrd** [comment]
Would you be able to take a quick look in the logs to see if you find any errors? This toast is a result of a failed call to the `list_remote_branches` tauri [command](https://github.com/gitbutlerapp/gitbutler/blob/9aa2115c122889a759cc25cac6cc227a97a6cc94/gitbutler-app/src/virtual_branches/commands.rs#L540).

How to find your log files: https://docs.gitbutler.com/development/debugging

**Comment by wesharper** [comment]
I think I've tracked down the relevant log:

```sh
2024-03-13T00:54:53.454395Z ERROR list_remote_branches: gitbutler-app/src/virtual_branches/errors.rs:809: list remote branches error error=failed to convert branches

Caused by:
    not found: the reference 'refs/remotes/origin/HEAD' cannot be peeled - Cannot resolve reference; class=Invalid (3); code=NotFound (-3) [3mproject_id[0m[2m=[0m"f33b1530-8095-4f45-886d-54b6beac5c16"
2024-03-13T00:54:53.454410Z  INFO list_remote_branches: gitbutler-app/src/virtual_branches/commands.rs:539: close time.busy=152ms time.idle=22.8µs [3mproject_id[0m[2m=[0m"f33b1530-8095-4f45-886d-54b6beac5c16"
2024-03-13T00:54:53.456310Z ERROR list_remote_branches: gitbutler-app/src/virtual_branches/errors.rs:809: list remote branches error error=failed to convert branches

Caused by:
    not found: the reference 'refs/remotes/origin/HEAD' cannot be peeled - Cannot resolve reference; class=Invalid (3); code=NotFound (-3) [3mproject_id[0m[2m=[0m"f33b1530-8095-4f45-886d-54b6beac5c16"
```

**Comment by wesharper** [comment]
I was able to resolve locally by deleting `.git/refs/remotes/origin/HEAD`.

**Comment by Hainesy** [comment]
I had exactly the same issue, with the same cause (mine was also pointing 'develop' which he had since changed to 'main')

**Comment by Byron** [comment]
I have submitted a PR with what should be a fix for this: #3584.
